flink-runtime/src/test/java/org/apache/flink/runtime/metrics/groups/AbstractMetricGroupTest.java

```diff
@@ -71,6 +76,17 @@ protected String getGroupName(CharacterFilter filter) {
 		registry.shutdown().get();
 	}
 
+	@Test
+	public void testGetAllVariablesWithExclusions() {
+		MetricRegistry registry = NoOpMetricRegistry.INSTANCE;
+
+		AbstractMetricGroup<?> group1 = new ProcessMetricGroup(registry, "host");
+		assertThat(group1.getAllVariables(), IsMapContaining.hasKey(ScopeFormat.SCOPE_HOST));
+
+		AbstractMetricGroup<?> group2 = new ProcessMetricGroup(registry, "host");
+		assertEquals(group2.getAllVariables(-1, Collections.singleton(ScopeFormat.SCOPE_HOST)).size(), 0);
```

Comment:
nit: One could phrase the assertion a bit nicer: `assertThat(group2.getAllVariables(-1, Collections.singleton(ScopeFormat.SCOPE_HOST)), isMapWithSize(0));` with `isMapWithSize` according to https://stackoverflow.com/a/36087146/4815083

Reply:
isMapWithSize doesn't exist in the hamcrest version we are using; otherwise I would've used it.

Reply:
Why not introducing `isMapWithSize` as a utility? I guess there are plenty of other places where such a utility would be helpful. Once we upgrade Hamcrest one could easily replace it. Now it will be super hard to find all locations where it would be applicable.